log-collector/src/main/scala/io/phdata/pulse/logcollector/LogCollectorCliParser.scala

```diff
@@ -19,7 +19,29 @@ package io.phdata.pulse.logcollector
 import org.rogach.scallop.ScallopConf
 
 class LogCollectorCliParser(args: Seq[String]) extends ScallopConf(args) {
-  lazy val port    = opt[Int]("port", required = true, descr = "Listening port")
+  lazy val port    = opt[Int]("port", required = false, descr = "HTTP Server Listening port")
   lazy val zkHosts = opt[String]("zk-hosts", required = true, descr = "Zookeeper hosts")
+  lazy val mode = opt[String]("consume-mode",
+                              required = false,
+                              descr = "'http' or 'kafka'",
+                              default = Some("http"))
+  lazy val kafkaProps =
+    opt[String]("kafka-properties", required = false, descr = "Kafka properties file")
+  lazy val topic = opt[String]("topic", required = false, descr = "Kafka Topic")
+
+  validateOpt(mode, port) {
+    case (Some("http") | None, None) => Left("Need a port if running http mode")
+    case _                           => Right(Unit)
+  }
+
+  validateOpt(mode, kafkaProps, topic) {
+    case (Some("kafka"), None, Some(_)) =>
+      Left("Need a kafka properties file if running kafka mode")
+    case (Some("kafka"), Some(_), None) => Left("Need a topic if running kafka mode")
```

Comment:
Can you make this more detailed, like `--topic argument needed`

Reply:
Fixed. Changed to "--topic argument needed if --consume-mode=kafka"